storage/storage_image.go

```diff
@@ -1130,6 +1111,25 @@ func (s *storageImageDestination) Commit(ctx context.Context, unparsedToplevel t
 		}
 		logrus.Debugf("saved image metadata %q", string(metadata))
 	}
+	// Set the reference's name on the image.  We don't need to worry about avoiding duplicate
+	// values because SetNames() will deduplicate the list that we pass to it.
+	if name := s.imageRef.DockerReference(); len(oldNames) > 0 || name != nil {
```

Comment:
FWIW not quite related to this PR:
- AFAICS the `len(oldNames) > 0` condition is unnecessary, we would only be setting names to the current value
- If we are worried about races between the call to `CreateImage` and this point, it might make sense to move the `oldNames` computation (which no longer needs to happen at this line) inside this conditional, to decrease (but not eliminate) a time window during which names added by a concurrent caller would be overwritten by `oldNames + name`, losing the other updates.
- … So, to be fully correct, there probably should be an `AddName` that does the update atomically under the c/storage lock, instead of `SetNames`.